src/psc_datalogger/connection.py

```diff
@@ -492,3 +509,41 @@ def _exit(self) -> None:
         # Note it will do 1 more iteration of the loop, inlcuding the sleep
         self.running = False
         self.logging_signal.set()
+
+    def _connection_check(self) -> None:
+        """Check whether connection is initialized.
+
+        Mostly required to guard against potential checks before the Prologix
+        controller has connected (or maybe there isn't even one plugged in yet!)"""
+        if not hasattr(self, "connection"):
+            logging.warning(
+                "Attempted to write to connection when it was not initialized"
+            )
+            raise ConnectionNotInitialized()
+
+    def _connection_write(self, command: str) -> None:
+        """Wrapper for all connection write commands."""
+        self._connection_check()
+        self.connection.write(command)
+
+    def _connection_bytes_in_buffer(self) -> int:
+        """Wrapper for accessing connection.bytes_in_buffer"""
+        self._connection_check()
+        return self.connection.bytes_in_buffer
+
+    def _connection_read(self) -> str:
+        """Wrapper for all connection read commands."""
+        self._connection_check()
+        return self.connection.read()
+
+    def _connection_timeout(self, timeout: int) -> None:
```

Comment:
I would use a verb for this, e.g. set_connection_timeout. At the moment this uses a noun like 'connection_bytes_in_buffer', except it's actually a function and behaves differently.

Reply:
Updated in 12288e0aab432a492f497cddbf445f9c2dd1fce5